src/main/kotlin/com/sainsburys/k2zpl/command/barcode/BarCode128.kt

```diff
@@ -0,0 +1,75 @@
+package com.sainsburys.k2zpl.command.barcode
+
+import com.sainsburys.k2zpl.builder.ZplBuilder
+import com.sainsburys.k2zpl.builder.toZplYesNo
+import com.sainsburys.k2zpl.command.ZplParameters
+import com.sainsburys.k2zpl.command.fieldData
+import com.sainsburys.k2zpl.command.fieldOrigin
+import com.sainsburys.k2zpl.command.fieldSeparator
+import com.sainsburys.k2zpl.command.options.ZplBarCode128Mode
+import com.sainsburys.k2zpl.command.options.ZplFieldOrientation
+import com.sainsburys.k2zpl.command.options.ZplYesNo
+import com.sainsburys.k2zpl.command.zplParameters
+
+internal data class BarCode128(
+    override val orientation: ZplFieldOrientation,
+    override val height: Int,
+    val line: ZplYesNo,
+    val lineAbove: ZplYesNo,
+    val checkDigit: ZplYesNo,
+    val mode: ZplBarCode128Mode
+) : BarCode() {
+
+    init {
+        require(height in 1..32000) { "Height must be between 1 and 32000" }
+    }
+
+    override val command = "^BC"
+
+    override val parameters: ZplParameters = zplParameters(
+        "o" to orientation.code,
+        "h" to height,
+        "l" to line,
+        "la" to lineAbove.toString(),
+        "c" to checkDigit.toString(),
+        "m" to mode.toString()
+    )
+}
+
+/**
+ * Creates a Code 128 barcode marker
+ * @param data data encoded in the barcode
+ * @param x horizontal position
+ * @param y vertical position
+ * @param orientation The orientation of the barcode.
+ * @param height The height of the barcode.
+ * @param interpretationLine print interpretation line
+ * @param lineAbove print interpretation line above code
+ * @param checkDigit UCC check digit
+ * @param mode barcode mode
+ */
+fun ZplBuilder.barcode128(
+    data: String,
+    x: Int,
+    y: Int,
+    orientation: ZplFieldOrientation = ZplFieldOrientation.NORMAL,
+    height: Int,
```

Comment:
I'm tempted to move `height` up with the other mandatory parameters so they're all one place.

If we didn't do this, I'm not sure how the Kotlin compiler would interpret things if we accidentally did the following:

```kotlin
BarCode39(
    "1234567890",
    10,
    10,
    10 // Does it know that this is the height?
)
```

What do you think?